extensions/sql-database-projects/src/models/project.ts

```diff
@@ -1482,12 +1592,82 @@ export class Project implements ISqlProject {
 		const firstPropertyElement = propertyElements[0];
 		if (firstPropertyElement.childNodes.length !== 1) {
 			// Property items are expected to have simple string content
-			throw new Error(constants.invalidProjectPropertyValue(propertyName));
+			throw new Error(constants.invalidProjectPropertyValueInSqlProj(propertyName));
 		}
 
 		return firstPropertyElement.childNodes[0].nodeValue!;
 	}
 
+	/**
+	 * Sets the value of project property, overrides value if already present in project file.
+	 * @param propertyName Name of property
+	 * @param propertyValue Value of property
+	 */
+	private async setProjectPropertyValue(propertyName: string, propertyValue: string): Promise<void> {
+		if (this.projFileXmlDoc === undefined) {
+			return;
+		}
+
+		const propertyGroups = this.projFileXmlDoc.getElementsByTagName(constants.PropertyGroup);
+		let propertyGroup = propertyGroups.length > 0 ? propertyGroups[0] : null;
+		if (propertyGroup === null) {
+			propertyGroup = this.projFileXmlDoc.createElement(constants.PropertyGroup);
+			this.projFileXmlDoc.documentElement?.appendChild(propertyGroup);
+		}
+
+		const propertyElements = propertyGroup.getElementsByTagName(propertyName);
+		let propertyElement: Element | undefined;
+		if (propertyElements.length === 0) {
+			propertyElement = this.addProjectPropertyTag(propertyName);
+			propertyElement?.appendChild(this.projFileXmlDoc.createTextNode(propertyValue));
+		} else {
+			propertyElement = propertyElements[0];
+			if (propertyElement.childNodes.length > 0) {
```

Comment:
Hm... what if there are no children nodes, i.e.
```xml
<MyProperty />
```
Does this mean we will never set a value for it?